Tests/GRPCCoreTests/Call/Client/ClientResponseTests.swift

```diff
@@ -182,3 +210,21 @@ final class ClientResponseTests: XCTestCase {
     }
   }
 }
+
+extension StreamingClientResponse.Contents.BodyPart: Equatable where Message: Equatable {
+  static func == (
+    lhs: StreamingClientResponse.Contents.BodyPart,
+    rhs: StreamingClientResponse.Contents.BodyPart
+  ) -> Bool {
+    switch (lhs, rhs) {
+    case (.message(let lhsMessage), .message(let rhsMessage)):
+      return lhsMessage == rhsMessage
+
+    case (.trailingMetadata(let lhsMetadata), .trailingMetadata(let rhsMetadata)):
+      return lhsMetadata == rhsMetadata
+
+    default:
+      return false
+    }
+  }
+}
```

Comment:
We should just make `BodyPart` and `Contents` conditionally `Hashable` when `Message` is `Hashable` (as public API).